build/Dockerfile

```diff
@@ -55,6 +55,9 @@ WORKDIR /build/windows-machine-config-operator/
 RUN make kubelet
 
 # Build kube-proxy
+WORKDIR /build/windows-machine-config-operator/kube-proxy/
+RUN git checkout tags/v1.30.0
```

Comment:
They are cloned in the midstream Dockerfile, it may be needed only there. However, we try to maintain parity between CI and midstream so we can detect issues pre-merge.

Reply:
Makes sense to me.

Reply:
@mansikulkarni96 I wasn't able to find where in midstream we clone the repos. Is it based on the submodule state of the [WMCO repo as defined in upstream_sources.yaml](https://gitlab.cee.redhat.com/openshift-winc-midstream/openshift-winc-midstream/-/blob/rhaos-4.17-rhel-9/upstream_sources.yml?ref_type=heads#L4)? If so then we shouldn't need any midstream changes

Reply:
yes, cachito clones them based on the state of the upstream commit.

Reply:
Changes LGTM, no need to update mdistream.